components/sts/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -288,6 +288,7 @@ private void process(HttpServletRequest request, HttpServletResponse response,
         reqToken.setPolicy(sessionDTO.getPolicy());
         reqToken.setPseudo(session.getId());
         reqToken.setUserName(authnResult.getSubject().getAuthenticatedSubjectIdentifier());
```

Comment:
This is the subject identifier set by the framework. This may not be the username. If we send this in our service call we could get unexpected errors. Instead we have to send the username which can be taken from the AuthenticatedUser object.

Also we have check all the places where tenant domain is extracted from the username in the backend and used. All those places we have to use the new tenant domain parameter instead.

Also if claims are fetched from the backend in passive sts component, we can stop that, since claims are handled in the framework.